x-pack/plugins/security_solution/server/usage/detections/detections_helpers.ts

```diff
@@ -301,3 +483,116 @@ export const getMlJobMetrics = async (
 
   return [];
 };
+
+export const getDetectionRuleMetrics = async (
+  kibanaIndex: string,
+  signalsIndex: string,
+  esClient: ElasticsearchClient,
+  savedObjectClient: SavedObjectsClientContract
+): Promise<DetectionRuleAdoption> => {
+  let rulesUsage: DetectionRulesTypeUsage = initialDetectionRulesUsage;
+  const ruleSearchOptions: RuleSearchParams = {
+    body: { query: { bool: { filter: { term: { 'alert.alertTypeId': SIGNALS_ID } } } } },
+    filterPath: [],
+    ignoreUnavailable: true,
+    index: kibanaIndex,
+    size: 10_000,
+  };
+
+  let detectionAlertsResp: AlertsAggregationResponse | undefined;
+
+  try {
+    const { body: ruleResults } = await esClient.search<RuleSearchResult>(ruleSearchOptions);
+
+    // @ts-expect-error `SearchResponse['hits']['total']` incorrectly expects `number` type instead of `{ value: number }`.
+    ({ body: detectionAlertsResp } = await esClient.search({
+      index: signalsIndex,
+      size: 0,
+      body: {
+        aggs: {
+          detectionAlerts: {
+            terms: { field: 'signal.rule.id.keyword' },
+          },
+        },
+        query: {
+          bool: {
+            filter: [
+              {
+                range: {
+                  '@timestamp': {
+                    gte: 'now-24h',
+                    lte: 'now',
+                  },
+                },
+              },
+            ],
+          },
+        },
+      },
+    }));
+
+    const cases = await savedObjectClient.find<CasesSavedObject>({
+      type: 'cases-comments',
+      fields: [],
+      page: 1,
+      perPage: 10_000,
```

Comment:
That's a neat little trick,  doing `10_000` 👍

I'm going to use that more now.